client/app/components/sections/Topbar/Topbar.js

```diff
@@ -121,6 +121,36 @@ const DEFAULT_CONTEXT = {
   loggedInUsername: null,
 };
 
+const SEARCH_PARAMS_TO_KEEP = ['view', 'locale'];
+const parseKeepParams = urlUtils.currySearchParams(SEARCH_PARAMS_TO_KEEP);
+const SEARCH_PARAMS = ['q', 'lq'];
+const parseSearchParams = urlUtils.currySearchParams(SEARCH_PARAMS);
+
+const isValid = (value) => typeof value === 'number' && !isNaN(value) || !!value;
```

Comment:
maybe more information about what kind of "valid" we're talking about? e.g. `isValidSearchParam`